src/modules/voltmeter/voltmeter.cpp

```diff
@@ -0,0 +1,135 @@
+#include <rtos-kochab.h>
+#include "boilerplate.h"
+#include "ros.hpp"
+#include "Publisher.hpp"
+#include "NodeHandle.hpp"
+#include "std_msgs/Float64.hpp"
+#include "adc.h"
+
+std_msgs::Float64 vm_buffer[5];
+ros_echronos::NodeHandle * volatile nh_ptr = NULL;
+ros_echronos::Publisher<std_msgs::Float64> * pub;
+
+#define SYSTICKS_PER_SECOND     100
+
+#define CAN_BITRATE 500000
+#define CAN_MSG_LEN 8
+
+static tCANMsgObject rx_object;
+static uint8_t can_input_buffer[CAN_MSG_LEN];
+
+static void init_can(void);
+static void write_can(uint32_t message_id, uint8_t *buffer, uint32_t buffer_size);
+
+extern "C" bool tick_irq(void) {
+    rtos_timer_tick();
+    return true;
+}
+
+bool sent_message;
+
+static uint32_t error_flag;
+
+static bool adc_cap;
+static uint32_t adc_buf[1];
+
+
+extern "C" void callback(void) {
+    adc_cap = true;
+}
+
+
+extern "C" void task_voltmeter_fn(void) {
+
+    // this creates a node handle
+    ros_echronos::ROS_INFO("Entered CAN task. Initializing...\n");
+    ros_echronos::NodeHandle nh;
+    nh.init("voltmeter_fn", "voltmeter_fn", RTOS_INTERRUPT_EVENT_ID_CAN_RECEIVE_EVENT, RTOS_SIGNAL_ID_CAN_RECEIVE_SIGNAL);
+    ros_echronos::ROS_INFO("Done init\n");
+    nh_ptr = &nh;
+
+    // publisher
+    ros_echronos::ROS_INFO("pub init\n");
+    ros_echronos::Publisher<std_msgs::Float64> _pub("voltmeter", (std_msgs::Float64*)vm_buffer, 5, false);

```

Comment:
probably should be `/voltmeter` also I `_pub` is a bit of a weird naming convertion

Reply:
Changed that to `/sensors/voltmeter` as that's already being used in the GUI